src/mudlet-lua/lua/geyser/GeyserMapper.lua

```diff
@@ -49,8 +87,26 @@ function Geyser.Mapper:new (cons, container)
 
   -----------------------------------------------------------
   -- Now create the Mapper using primitives
-  createMapper(me.windowname, me:get_x(), me:get_y(),
-  me:get_width(), me:get_height())
+  if me.dockPosition and me.dockPosition:lower() == "floating" then
+    me.dockPosition = "f"
+  end
+
+  if not me.mapWidget and not me.docked and not me.dockPosition then
+    createMapper(me.windowname, me:get_x(), me:get_y(),
+    me:get_width(), me:get_height())
+  else
+    me.mapWidget = true
```

Comment:
Seems to be a little bit of a bug around the mapWidget - it's possible to spawn two map widgets, one embedded and one floating, by doing:

```lua
mapWidget = Geyser.Mapper:new({mapWidget = false})
```

And then changing it to `true` and re-running the script.

Could I recommend a name better than `mapWidget`? It's not intuitive to remember whenever false or true means floating or embedded state. How about using a string "floating" or "embedded", and defaulting to "embedded" as the previous API did?

Reply:
> Seems to be a little bit of a bug around the mapWidget - it's possible to spawn two map widgets, one embedded and one floating, by doing:

> mapWidget = Geyser.Mapper:new({mapWidget = false})

And then changing it to true and re-running the script.

Thats not really a new bug (or feature ;) depends how you see it). You can do the same already in 4.6.1 by first open a map window by clicking on the map icon and then use createMapper() for creating a embedded mapper.

> Could I recommend a name better than mapWidget? It's not intuitive to remember whenever false or true means floating or embedded state. How about using a string "floating" or "embedded", and defaulting to "embedded" as the previous API did?

I agree with you ( I'm not good at variable names) 
How about using a boolean embedded which is normally true and if I set that one to false at creation it creates the map window and if it's true (or not set) the embedded mapper.
+ I think I will get rid of docked as it is just doing the exactly same as dockPosition ="right".
I just added it because I was using it also for the UserWindow.

Reply:
Yeah, let's try those suggestions out and see how they feel.